### PR TITLE
drivers: xiphera_trng: Add support for XIP8001B TRNG

### PR DESCRIPTION
**Pull request by vesajaaskelainen**
Adds support for Xiphera's XIP8001B true random number generator.

XIP8001B is a FPGA IP core that can be synthesized in FPGA devices to
provide TRNG source for device where it is missing like Xilinx Zynq-7000
and Xilinx Zynq MPSoC.

Signed-off-by: Vesa Jääskeläinen <vesa.jaaskelainen@vaisala.com>
Co-developed-by: Atte Tommiska <atte.tommiska@xiphera.com>

---

Example device tree entries:
```
       xiphera_trng_0: xiphera_trng@aaaa0000 {
                compatible = "xiphera,xip8001b-trng";
                status = "okay";
                secure-status = "disabled";

                reg = <0x0 0xaaaa0000 0x0 0x10000>;

                clock-names = "S_AXI_ACLK";
                clocks = <&zynqmp_clk 71>;
        };

        xiphera_trng_1: xiphera_trng@bbbb0000 {
                compatible = "xiphera,xip8001b-trng";
                status = "disabled";
                secure-status = "okay";

                reg = <0x0 0xbbbb0000  0x0 0x10000>;

                clock-names = "S_AXI_ACLK";
                clocks = <&zynqmp_clk 71>;
        };
```

Complementary PR:
- https://github.com/OP-TEE/optee_os/pull/5350

### CODE REVIEW

File: core/drivers/xiphera_trng.c [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+static TEE_Result xiphera_trng_probe(const void *fdt, int node,
+				     const void *compat_data __unused)
+{
+	int dt_status = _fdt_get_status(fdt, node);
+	uint32_t status = 0;
+	size_t size = 0;
+
+	/* Skip non-secure instances */
+	if (dt_status != DT_STATUS_OK_SEC)
+		return TEE_ERROR_GENERIC;
```
Comment (vesajaaskelainen) [inline core/drivers/xiphera_trng.c:117]: Not this has side effect of:
```
E/TC:0 0 probe_driver_node:364 Failed to probe xiphera_trng on node xiphera_trng@a3f00000: 0xffff0000
I/TC: HUK ready
E/TC:0 0 probe_dt_drivers:701 Probe sequence result: 0
E/TC:0 0 print_probe_list:320 |- Driver xiphera_trng on node xiphera_trng@a3f00000 failed
I/TC: Primary CPU switching to normal world boot
```
We are using system device tree that defines more devices than what just OP-TEE needs.

We have dedicated TRNG instance for Linux and one for OP-TEE.

I used similar method for this return value as what was in existing driver but I would like to silence the error message as it is not really a error message. Any advice on this?

**Comment by attetommiska** [comment]
`Signed-off-by: Atte Tommiska <atte.tommiska@xiphera.com>`

File: core/drivers/xiphera_trng.c (hunk 1 above)
Comment (jenswi-linaro) [reply core/drivers/xiphera_trng.c:117]: It looks like we may need a `TEE_ERROR_NODE_DISABLED` error code.

Comment (vesajaaskelainen) [reply core/drivers/xiphera_trng.c:117]: OK. Let's see how that unravels.

Comment (vesajaaskelainen) [reply core/drivers/xiphera_trng.c]: Added commit for the feature. Please have a look.

Now output is silent on this matter.

**Comment by vesajaaskelainen** [comment]
> `Signed-off-by: Atte Tommiska <atte.tommiska@xiphera.com>`

Applied tag. Thanks!

**Comment by vesajaaskelainen** [comment]
Rebased on top of latest master.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by vesajaaskelainen** [comment]
> `Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

Applied tag. Thanks!

**Pull request merged**